### ISSUE TITLE
Documentation progress

### ISSUE THREAD

**Issue by pi0** [nuxt/framework#4104]
(This list is WIP. Credits to @Atinux and @clemcode for preparation)

# Docs

## Completion overview

This section is used to track progress on the documentation. Feel free to edit if you complete a section.

### Getting started

Category        | Title          | Status
----------------|----------------|------------------
Getting started | Quick start    | OK
Getting started | Tutorial       | In Review
Migration       | Summary        | TODO
Migration       | Bridge         | OK
Migration       | V3             | OK

### Guide

Category        | Title          | Status
----------------|----------------|------------------
Concepts        | What is Nuxt?  | Rework
Concepts        | Vue.js development  | OK
Concepts        | Rendering modes  | OK
Concepts        | Auto imports  | OK
Concepts        | Server engine  | OK
Concepts        | Typescript  | OK
Features        | Views  | TODO
Features        | Routing  | TODO
Features        | Assets  | TODO
Features        | Head Management  | Rework
Features        | Data Fetching  | Rework
Features        | State management  | Rework
Features        | Error handling  | OK
Features        | Plugins  | TODO
Features        | API Routes  | TODO
Features        | Runtime config  | OK
Features        | Theming | TODO
Features        | Modules | TODO
Going Further   | Tooling | TODO
Going Further   | Testing | In progress
Going Further   | Native ES Modules | OK
Going Further   | Module creation | OK
Deployment   | Guide | TODO
Deployment   | Azure | OK
Deployment   | Cloudflare | OK
Deployment   | Firebase | OK
Deployment   | Netlify | OK
Deployment   | PM2 | OK
Deployment   | Vercel | OK
Deployment   | Presets | OK

### API Reference

Category        | Title          | Status
----------------|----------------|------------------
Composables     | useAsyncData   | TODO
Composables     | useCookie      | TODO
Composables     | useError       | TODO
Composables     | useFetch       | TODO
Composables     | useHydration   | TODO
Composables     | useMeta        | TODO
Composables     | useNuxtApp     | TODO
Composables     | useRequestHeaders     | TODO
Composables     | useRoute      | TODO
Composables     | useRouter     | TODO
Composables     | useState      | TODO
Components      | NuxtPage      | TODO
Components      | NuxtLayout    | TODO
Components      | NuxtLink      | Rework
Components      | NuxtErrorBoundary | TODO
Utils           | $fetch      | TODO
Utils           | abortNavigation | TODO
Utils           | addRouteMiddleware | TODO
Utils           | clearError | TODO
Utils           | defineNuxtComponent | TODO
Utils           | defineNuxtRouteMiddleware | TODO
Utils           | definePageMeta | TODO
Utils           | navigateTo | TODO
Utils           | refreshNuxtData | TODO
Utils           | throwError | TODO
Directory Structure | .nuxt | To complete
Directory Structure | .output | To complete
Directory Structure | assets | OK
Directory Structure | components | OK
Directory Structure | composables | OK
Directory Structure | layouts | OK
Directory Structure | middleware | OK
Directory Structure | node_modules | OK
Directory Structure | pages | OK
Directory Structure | plugins | OK
Directory Structure | public | OK
Directory Structure | server | OK
Directory Structure | .gitignore | OK
Directory Structure | app.vue | OK
Directory Structure | nuxt.config.ts | OK
Directory Structure | .nuxtignore | OK
Directory Structure | package.json | OK
Directory Structure | tsconfig.json | OK
Server | Api | TODO
Server | Middleware | TODO
Server | Deployment presets | TODO
Advanced | Nuxt Internals | OK
Advanced | Nuxt Kit | OK
Advanced | Lifecycle hooks | OK

**Comment by Intevel** [comment]
Composables: useMeta, got renamed

**Comment by mwohlan** [comment]
I think useRuntimeConfig is missing in the API References.

**Comment by Atinux** [comment]
Added to the table, thanks @mwohlan

**Comment by baixiaoyu2997** [comment]
hi,thanks nuxt team work, does the nuxt3 doc have `nuxt lifecycle`? i mean like [nuxt2](https://nuxtjs.org/docs/concepts/nuxt-lifecycle/#client)

this is clear for understanding the nuxt life cycle

**Comment by Atinux** [comment]
Not yet but shall be documented indeed @baixiaoyu2997

**Comment by marcelxpfeifer** [comment]
Hey everyone, I guess the current docs are also missing the static deployment?

**Comment by pi0** [comment]
> Hey everyone, I guess the current docs are also missing the static deployment

This feature is still work in progress. But for sure! We plan to add deployment guide for static and generic Node server.

**Comment by jpsc** [comment]
Currently, all deployment documentation is failing. Shall I create a separate issue or is this a good place to report it?
https://v3.nuxtjs.org/guide/deployment/vercel

**Comment by rguttersohn** [comment]
Can deployment for Heroku be added to the table? I just deployed a nuxt 3 app on the platform and can contribute to the documentation for it.

**Comment by Atinux** [comment]
Add a note for experimental section to cover the bleeding edge features coming.

**Comment by noopurphalak** [comment]
Hi @pi0, It would be good to add a little note about a substitute for `nuxtServerInit` in nuxt 3 in the documentation.

**Comment by manniL** [comment]
@noopurphalak Use a server-side Nuxt plugin and you are good to go.

**Comment by noopurphalak** [comment]
@manniL Agreed, but I think it is important to make a note of it so that people are aware...

**Comment by hacknug** [comment]
I think `error.vue` should be added to the Directory Structure section on the Guide and from there link to the existing error-handling related sections that exist now.

**Comment by Draichi** [comment]
hello, the `Lifecycle hooks` item in the table has a OK status but in the docs page it [looks incomplete](https://nuxt.com/docs/api/advanced/hooks#nuxt-hooks-build-time). 

I say this because I want to help with the documentation, I just want to make sure it's necessary
